sap_business_document_processing/document_information_extraction_client/dox_api_client.py

```diff
@@ -0,0 +1,608 @@
+        """
+        Deletes the enrichment data records with the given IDs in the payload
+        :param client_id: The client ID for which the enrichment data was created
+        :param data_type: The type of enrichment data that should be deleted. Can be given as :enum:`DoxDataType` or
+        string. Options are 'businessEntity' or 'employee'
+        :param payload: A list of dictionaries with the form: ``{'id':'', 'system':'', 'companyCode':''}``
+        :param subtype: (optional) The subtype of the records that should be deleted. Only used for type
+        'businessEntity'. Can be given as instance of :enum:`DoxDataSubType` or as string. Options are 'supplier',
+        'customer' or 'companyCode'
+        :param delete_async: Set to ``True`` to delete data records asynchronously. Asynchronous deletion should be
+        used when deleting large amounts of data to improve performance. Default is ``False``
+        :return: The API endpoint response as dictionary
+        """
+        params = {
+            API_FIELD_CLIENT_ID: client_id,
+            API_REQUEST_FIELD_ENRICHMENT_TYPE: str(data_type)
+        }
+        if subtype is not None:
+            params[API_REQUEST_FIELD_ENRICHMENT_SUBTYPE] = str(subtype)
+        delete_url = DATA_ASYNC_ENDPOINT if delete_async else DATA_ENDPOINT
+        response = self.delete(delete_url, json={API_FIELD_VALUE: payload}, params=params,
+                               log_msg_before=f"Start deleting {len(payload) if len(payload) > 0 else 'all'} "
+                               f"enrichment data records for client {client_id}")
+
+        if delete_async:
+            job_id = response.json()[API_FIELD_ID]
+            response = self._poll_for_url(DATA_ID_ENDPOINT.format(id=job_id),
+                                          get_status=lambda r: r[API_FIELD_VALUE][API_FIELD_STATUS],
+                                          log_msg_after=f"Successfully deleted {len(payload) if len(payload) > 0 else 'all'} "
+                                          f"enrichment data records for client {client_id}")
+        return response.json()
+
+    def activate_enrichment_data(self):
+        """
+        Activates all enrichment data records for the current tenant
+        :return: The API endpoint response as dictionary
+        """
+        response = self.post(DATA_ACTIVATION_ASYNC_ENDPOINT,
+                             log_msg_before='Start activating enrichment data records')
+
+        response = self._poll_for_url(DATA_ACTIVATION_ID_ENDPOINT.format(id=response.json()[API_FIELD_ID]),
+                                      get_status=lambda r: r[API_FIELD_VALUE][API_FIELD_STATUS],
+                                      log_msg_after='Successfully activated enrichment data records')
+        return response.json()
+
+    def get_enrichment_data_activation_job(self, data_activation_job_id):
```

Comment:
Why do we require this as a separate public method?

Reply:
It is a leftover of a previous implementation of the enrichment data activation process. You are right, that it is not needed anymore.

Reply:
Thanks, let's remove.